qt/python/mantidqtinterfaces/mantidqtinterfaces/dns_powder_tof/file_selector/file_selector_presenter.py

```diff
@@ -0,0 +1,281 @@
+        self.watcher.sig_files_changed.connect(self._files_changed_by_watcher)
+
+    # loading
+    def _read_all(self, filtered=False, watcher=False, start=None, end=None):
+        """
+        Reading of new files, if filtered is True, only the files in the
+        range specified by start and end.
+        """
+        fn_range = [start, end]
+        data_path = self.param_dict['paths']['data_dir']
+        nb_files, loaded, datafiles, fn_range = \
+            self.model.set_datafiles_to_load(data_path, fn_range, filtered,
+                                             watcher)
+
+        self.view.open_progress_dialog(nb_files)
+        self.model.read_all(datafiles, data_path, loaded, watcher)
+        self.view.set_first_column_spanned(self.model.get_scan_range())
+        self._filter_scans()
+        if self.model.get_number_of_scans() == 1:
+            self.view.expand_all()
+
+        # re-adjust view to column width
+        self.view.adjust_treeview_columns_width(self.num_columns)
+
+    def _read_standard(self, self_call=False):
+        """
+        Reading of standard files.
+        """
+        data_path = self.param_dict['paths']['data_dir']
+        standard_path = self.param_dict['paths']['standards_dir']
+        if not standard_path:
+            self.raise_error('No path set for standard data')
+        standard_found = self.model.read_standard(standard_path)
+        self.view.set_first_column_spanned(self.model.get_scan_range())
+        self._filter_standard()
+        if not standard_found and not self_call:
+            if self.model.try_unzip(data_path, standard_path):
+                self._read_standard(self_call=True)
+        self.view.expand_all()
+        # re-adjust view to column width
+        self.view.adjust_treeview_columns_width(self.num_columns)
+
+    def _cancel_loading(self):
+        self.model.set_loading_canceled(True)
+
+    def _set_start_end(self, fn_range):
+        start, end = fn_range
+        self.view.set_start_end_file_numbers_from_arguments(start, end)
+
+    def _files_changed_by_watcher(self):
+        """
+        Triggered by view if new files are found and timer of 5 s is run
+        down.
+        """
+        self._read_all(watcher=True)
+
+    def _autoload_new(self, state):
+        data_dir = self.param_dict['paths']['data_dir']
+
+        if not data_dir:
+            self.raise_error('No data directory selected', critical=True)
+
+        if state == 2 and data_dir:
+            self.watcher.start_watcher(data_dir)
+            if not self._old_data_set:
+                self._read_all()
+        else:
+            self.watcher.stop_watcher()
+
+    def _auto_select_standard(self, state):
+        if state == 2:
+            self._check_all_visible_scans()
+
+    # scan selection
+    def _automatic_select_all_standard_files(self):
+        was = not self.model.model_is_standard()
+        if was:  # if view is not standard we change to it and change back
```

Comment:
Why is this required

Reply:
Thanks, it is not necessary since process_request() calls this method it in the "sample mode" anyways.